src/components/templates/PartyMap/components/Map/PartyMapRoomOverlay.tsx

```diff
@@ -14,104 +17,205 @@ import RoomAttendance from "../RoomAttendance";
+    dispatch(retainAttendance(true));
   }, [dispatch, room]);
 
-  const roomUnhovered = useCallback(() => {
-    dispatch(retainAttendance(false));
+  const handleRoomUnhovered = useCallback(() => {
     setRoomHovered(undefined);
+    dispatch(retainAttendance(false));
   }, [dispatch]);
 
-  const left = room.x_percent;
-  const top = room.y_percent;
-  const width = room.width_percent;
-  const height = room.height_percent;
+  // TODO: this is being spread here because the below code modifies it with .push()/etc
+  // const rooms = [...venue.rooms];
+
+  // TODO: what is the intent/purpose of this?
+  // TODO: how do we implement this nicely, and where?
+  // TODO: reinstate this functionality
+  // if (roomHovered) {
+  //   const idx = rooms.findIndex((room) => room.title === roomHovered.title);
+  //   if (idx !== -1) {
+  //     const chosenRoom = rooms.splice(idx, 1);
+  //     rooms.push(chosenRoom[0]);
+  //   }
+  // }
+
+  // TODO: what is a better semantic name for this?
+  // const onJoinRoom: React.MouseEventHandler = useCallback(
+  //   (e) => {
+  //     e.stopPropagation();
+  //
+  //     if (isExternalUrl(room.url)) {
+  //       openRoomUrl(room.url);
+  //     } else {
+  //       window.location.href = getRoomUrl(room.url);
+  //     }
+  //
+  //     enterCampRoom(room);
+  //   },
+  //   [enterCampRoom, room]
+  // );
+
+  // TODO: should this also openRoomUrl as above? If so, let's use the same function for it.. if not, why?
+  // TODO: what is a better semantic name for this?
+  // const onClick3 = useCallback(() => enterCampRoom(room), [
+  //   enterCampRoom,
+  //   room,
+  // ]);
+
+  // TODO: what is a better semantic name for this?
+  const shouldShowHovered =
+    venue.roomVisibility === RoomVisibility.hover &&
+    isRoomHovered &&
+    roomHovered?.title === room.title;
+
+  // TODO: what is a better semantic name for this?
+  const shouldShow1 =
+    !venue.roomVisibility || venue.roomVisibility === RoomVisibility.nameCount;
+
+  // TODO: what is a better semantic name for this?
+  const shouldShow2 =
+    shouldShow1 ||
+    (venue.roomVisibility === RoomVisibility.count && hasAttendance);
+
+  const containerStyles = useMemo(
+    () => ({
+      left: `${room.x_percent}%`,
+      top: `${room.y_percent}%`,
+      width: `${room.width_percent}%`,
+      height: `${room.height_percent}%`,
+    }),
+    [room.height_percent, room.width_percent, room.x_percent, room.y_percent]
+  );
 
   return (
     <div
       className={`room position-absolute ${isSelectedRoom && "isUnderneath"}`}
-      style={{
-        left: left + "%",
-        top: top + "%",
-        width: width + "%",
-        height: height + "%",
-      }}
-      key={room.title}
-      onClick={toggleRoom}
-      onMouseEnter={roomHovered}
-      onMouseLeave={roomUnhovered}
+      style={containerStyles}
+      // TODO: is this unique? Do we even need a key here since we're not directly inside a loop?
+      // key={room.title}
```

Comment:
- [ ] [TODO]: is this unique? Do we even need a key here since we're not directly inside a loop?

Reply:
Room titles are not unique.. we need to use something better than this. Ideally implementing `uuid` on rooms on the the backend.

Reply:
resolved in https://github.com/sparkletown/sparkle/pull/840/commits/4b6a1669636861e45d8f7344d9ba82a8437b29c8